---
layout: post
title:  "Genomics of Rare Disease Conference"
---


Attending the [*Genomics of Rare Disease* conference](https://coursesandconferences.wellcomeconnectingscience.org/event/genomics-of-rare-disease-20250409/) (9-11th April 2025, Hinxton Conference Centre, UK) was both inspiring and thought-provoking. Keynote-speaker Stephen Kingsmore [1] set the tone, highlighting how rapid WGS is transforming the care of critically ill infants, delivering diagnoses quickly and cost-effectively, especially in health systems like Medicaid. I also really liked the practical benchmark study by Kaschta and colleagues [2] in German hospital settings, focusing on which sequencing technology will deliver the best cost-effective diagnosis. It was reassuring to see how pipelines, speed/automation, and clinical utility intersect in real-world healthcare.

Throughout the sessions, there was a clear focus on scaling up genomic diagnostics. From Israel’s rapid newborn sequencing initiative presented by Hagit Baris-Feldman, to the UK’s ambitious *Generation Study* aiming to sequence 100,000 babies (talk by Dalia Kasperaviciute [3]), we’re seeing major investment in early-life genomics. Diagnoses are at the receiving end of rare variant detection, many talks  showcased neat methods to improve this field: long-read sequencing, structural variant detection, deep learning to investigate function. 

Therapeutically, case studies of N=1 drug repurposing (i.e. demonstrating that a known drug can be safely and successfully applied to another disease) and CRISPR/AAV-based targeting of regulatory elements were compelling. As a rare disease bioinformatician and aunty to a Batten disease niece, I was captivated by stories to shorten the diagnostic odyssey.

International perspectives added valuable context. Australia is using automated reanalysis to tackle diagnostic backlogs [4], Sweden hosted a multiomics hackathon to solve rare pediatric cases, meanwhile Saudi Arabia and Brazil are creatively navigating challenges around consanguinity and resource limitations. What's obvious to me is the lack of international collaboration for inclusive, cross-border genomic efforts.

We need accurate, fast, and interoperable tools that can support clinical diagnostics at scale. Equally important is building better, richer variant interpretation frameworks and ensuring functional data keeps pace with sequencing outputs. It’s an exciting time to be working at the intersection of data, biology, and healthcare—there’s a lot to do, but even more to look forward to.

_Read more:_
1: Kingsmore SF, Wright M, Smith LD, et al. Prequalification of genome-based newborn screening for severe childhood genetic diseases through federated training based on purifying hyperselection. Am J Hum Genet. 2024;111(12):2618-2642. doi:10.1016/j.ajhg.2024.10.021
2: Kaschta, D et al. Evaluating Genome Sequencing Strategies: Trio, Singleton, and Standard Testing in Rare Disease Diagnosis. medRxiv 2024.12.20.24319228
3: Kaplanis et al. Assessment of the variant prioritisation strategy for genomic newborn screening in the Generation Study. doi: 10.1101/2025.03.12.25323745 
4: Best, Stephanie, et al. Reanalysis of genomic data in rare disease: current practice and attitudes among Australian clinical and laboratory genetics services. European Journal of Human Genetics (2024): 1-8.
